ngx-fudis/projects/ngx-fudis/src/lib/components/form/fieldset/fieldset.component.spec.ts

```diff
@@ -1,34 +1,174 @@
+
+@Component({
+  selector: 'fudis-mock-fieldset-component',
+  template: ` <fudis-fieldset
+    [title]="'Fieldset title'"
+    [helpText]="'Fieldset help text'"
+    [required]="required"
+  >
+    <ng-template fudisActions [type]="'fieldset'">
+      <p class="test-actions-content">This is actions content</p>
+    </ng-template>
+    <ng-template fudisNotifications [type]="'fieldset'">
+      <p class="test-notifications-content">This is notifications content</p>
+    </ng-template>
+    <ng-template fudisContent [type]="'fieldset'">
+      <p class="test-fieldset-content">This is fieldset content</p>
+    </ng-template>
+    <p class="test-do-not-find">You should not find me</p>
+  </fudis-fieldset>`,
+})
+class MockFieldSetComponent {
+  required = false;
+
+  fieldsetExample = new FormGroup({
+    exampleTextInput: new FormControl(null, FudisValidators.required('This field is required')),
+  });
+}
 
 describe('FieldSetComponent', () => {
-  let component: FieldSetComponent;
-  let fixture: ComponentFixture<FieldSetComponent>;
+  let componentMock: MockFieldSetComponent;
+  let fixtureMock: ComponentFixture<MockFieldSetComponent>;
+  let fieldsetElement: HTMLFieldSetElement;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [
+        ActionsDirective,
+        BodyTextComponent,
+        ButtonComponent,
+        ContentDirective,
         FieldSetComponent,
-        MockComponent(GridComponent),
+        GridComponent,
         GridApiDirective,
         GridDirective,
+        GuidanceComponent,
+        IconComponent,
+        LabelComponent,
+        MockFieldSetComponent,
+        NotificationComponent,
+        NotificationsDirective,
+        TextInputComponent,
+        ValidatorErrorMessageComponent,
       ],
       providers: [FudisGridService, FudisInternalErrorSummaryService, FudisBreakpointService],
+      imports: [ReactiveFormsModule],
     }).compileComponents();
 
-    fixture = TestBed.createComponent(FieldSetComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+    fixtureMock = TestBed.createComponent(MockFieldSetComponent);
+    componentMock = fixtureMock.componentInstance;
+    fixtureMock.detectChanges();
+  });
+
+  describe('Wrapper fieldset element HTML attributes', () => {
+    beforeEach(() => {
+      fieldsetElement = getElement(fixtureMock, 'fieldset') as HTMLFieldSetElement;
+    });
+
+    it('should have native fieldset element present', () => {
+      expect(fieldsetElement).toBeTruthy();
+    });
+
+    it('should have legend element as a first child', () => {
+      expect(fieldsetElement.firstChild?.nodeName).toEqual('LEGEND');
+    });
+
+    it('should have id constructed through IdService', () => {
+      expect(fieldsetElement.id).toEqual('fudis-fieldset-1');
+    });
+
+    it('should have default CSS class for fudis-fieldset', () => {
+      expect(fieldsetElement.className).toContain('fudis-fieldset');
+    });
+
```

Comment:
Tänne voisi vielä kirjoittaa testi inputSize inputille

Reply:
The use of this could be also documented in Storybook, but since this is used mostly in radio-button group and checkbox-group. This documentation should not be exposed.